src/main/java/core/basesyntax/utils/Figures.java

```diff
@@ -0,0 +1,39 @@
+package core.basesyntax.utils;
+
+import core.basesyntax.abstraction.Figure;
+import core.basesyntax.figures.Circle;
+import core.basesyntax.figures.IsoscelesTrapezoid;
+import core.basesyntax.figures.Rectangle;
+import core.basesyntax.figures.RightTriangle;
+import core.basesyntax.figures.Square;
+import java.util.Random;
+
+public enum Figures {
+    SQUARE,
+    RECTANGLE,
+    RIGHT_TRIANGLE,
+    CIRCLE,
+    ISOSCELES_TRAPEZOID;
+
+    private static final int MAX_VALUE = 100;
+
+    public Figure getFigure(Random random, ColorSupplier colorSupplier) {
+        String color = colorSupplier.getRandomColor();
+
+        switch (this) {
+            case SQUARE:
+                return new Square(color, random.nextInt(MAX_VALUE));
+            case RECTANGLE:
+                return new Rectangle(color, random.nextInt(MAX_VALUE),
+                        random.nextInt(MAX_VALUE));
+            case RIGHT_TRIANGLE:
+                return new RightTriangle(color, random.nextInt(MAX_VALUE),
+                        random.nextInt(MAX_VALUE));
+            case CIRCLE:
+                return new Circle(color, random.nextInt(MAX_VALUE));
+            default:
+                return new IsoscelesTrapezoid(color, random.nextInt(MAX_VALUE),
+                        random.nextInt(MAX_VALUE), random.nextInt(MAX_VALUE));
+        }

```

Comment:
enums should not contain business logic. Let's move all this logic to FigureSupplier and create a private nested enum